packages/wonder-blocks-data/src/hooks/use-cached-effect.ts

```diff
@@ -265,29 +262,36 @@ export const useCachedEffect = <TData extends ValidCacheData>(
         }
         fetchRequest();
         return () => {
-            // @ts-expect-error [FEI-5019] - TS2339 - Property 'cancel' does not exist on type 'never'.
             currentRequestRef.current?.cancel();
-            // @ts-expect-error [FEI-5019] - TS2322 - Type 'null' is not assignable to type 'undefined'.
             currentRequestRef.current = null;
         };
     }, [shouldFetch, fetchRequest]);
 
     // We track the last result we returned in order to support the
-    // "retainResultOnChange" option.
-    const lastResultAgnosticOfIdRef = React.useRef(Status.loading());
+    // "retainResultOnChange" option. To begin, the last result is no-data.
+    const lastResultAgnosticOfIdRef = React.useRef<Result<TData>>(
+        Status.noData<TData>(),
+    );
+    // The default return value is:
+    // - The last result we returned if we're retaining results on change.
+    // - The no-data state if shouldFetch is false, and therefore there is no
+    //   in-flight request.
+    // - Otherwise, the loading state (we can assume there's an inflight
+    //   request if skip is not true).
     const loadingResult = retainResultOnChange
         ? lastResultAgnosticOfIdRef.current
-        : Status.loading();
+        : shouldFetch
+        ? Status.loading<TData>()
+        : Status.noData<TData>();
```

Comment:
**note:** Here we default our "loadingResult" to no-data if we aren't fetching (which encompasses both the hard skip of the skip prop but also states where we won't fetch at all, such as the CacheOnly fetch policy).